### DISCUSSION TITLE
"Consensus Leader - A in term 5" with high CPU/Memory usage in Single-Node-Cluster

### DISCUSSION THREAD

**Discussion by oiBio** [ravendb/ravendb#16710] [Q&A] [answered]
I have a Single Node Cluster and noticed a high CPU usage, while the server just served low requests per second.
 
Unter Advanced i noticed this
![image](https://github.com/ravendb/ravendb/assets/33127462/b1177157-a832-420c-a759-d7aa927a4d12)
![image](https://github.com/ravendb/ravendb/assets/33127462/36c6a763-9c9f-4b09-9239-e704d727c0be)

It is a Single Node, and always has been. I think there is a bug or a loop, there is no way this should use this much CPU/Memory. I uses one Core with 100% and does not end.

I am pretty sure i could "fix" this when i restart the node, but i would love to help fixing this bug.
I am using 5.4.104

**Comment by grisha-kotler** [comment]
Can you go to press the stack trace button, so we can see what it's doing?

**Comment by oiBio** [comment]
sure :-)
[stacks.zip](https://github.com/ravendb/ravendb/files/11815914/stacks.zip)

**Comment by karmeli87** [comment]
Hi,
This is indeed very strange.

What kind of cluster operation do you process? 
Could you share the output of the following EPs
`/admin/debug/cluster/history-logs`
`/admin/cluster/log`

Also, what is the size of the system folder?

**Comment by oiBio** [comment]
And here are the logs.
[logs.zip](https://github.com/ravendb/ravendb/files/11815931/logs.zip)

You mean this?
![image](https://github.com/ravendb/ravendb/assets/33127462/5c92142e-3395-49e7-9cf0-ca07f9b5426e)

**Reply by karmeli87** [reply]
Yes and thanks for the logs, I think I have a lead.

Are you doing a constant import?
If it isn't a production instance, could you pause the client not to send anything to the server (especially the import)  and check whether it goes down?

**Reply by oiBio** [reply]
Ah yes you are right. i imported some indexes and compare exchanges - via ravendbdump - this morning and forgot about ist. 
Its still running. There should be around 14k compare exchange keys.

![image](https://github.com/ravendb/ravendb/assets/33127462/00469bd2-8e2f-40af-9c37-220fdbeeb0bd)

i just aborted this
![image](https://github.com/ravendb/ravendb/assets/33127462/287acadc-b10b-4623-b83e-1996f1056b75)

and CPU usage got even worse :-/ now its 100%
![image](https://github.com/ravendb/ravendb/assets/33127462/45518f7f-3dde-4fd3-9303-afb2181229c2)

![image](https://github.com/ravendb/ravendb/assets/33127462/320619c2-cd7d-44df-8f0e-7b915219ed13)

[newlogs.zip](https://github.com/ravendb/ravendb/files/11816163/newlogs.zip)

**Reply by karmeli87** [reply]
Not sure about the high CPU at this point, but here is the memory issue https://issues.hibernatingrhinos.com/issue/RavenDB-20742 :)

**Reply by oiBio** [reply]
good :-)
the cpu is still at 100%, should i restart the server or can i provide you some more data so you can find this issue too?

**Reply by karmeli87** [reply]
Would love to figure out the high CPU as well.. can you send us few more stack traces of that thread?

**Reply by karmeli87** [reply]
Also, can you open the admin logs on that node and mark `Show Error Only` to check if you see any error there

**Reply by oiBio** [reply]
![image](https://github.com/ravendb/ravendb/assets/33127462/266b05e3-6ce7-4307-8491-b7fd35a85042)

[RavenDbLogs.zip](https://github.com/ravendb/ravendb/files/11816470/RavenDbLogs.zip)

**Reply by karmeli87** [reply, marked as answer]
Thanks a lot! this was very helpful and I think we have all the info needed.
Here I believe is the high CPU issue
https://issues.hibernatingrhinos.com/issue/RavenDB-20744/High-CPU-in-a-single-leader-cluster

**Reply by oiBio** [reply]
nice, i restarted my server and i am back at normal
